### PR TITLE
User and Admin Signup/Signin/Login Page

### PR DESCRIPTION
**Pull request by curtydudes**
This code address the Trader -- User Story 1&2. I also added here the RSPEC Model for users. 

I set up multiple devise accounts following the wiki here for devise: https://github.com/heartcombo/devise/wiki/How-to-Setup-Multiple-Devise-User-Models. Following this, I will create a function where the Admin can only login their credentials as the sign up will happen in the console as per the wiki. 

Models: 
I got to create an admin and user models.

Views: 
Devise automatically created admin and user views while I manually added a landing folder with a temporary landing page called signupUser.html.rb (will be changing this name appropriately)

Controllers: 
Created a landing page temporarily to set up shop and to have some pseudo base
An admins controller and users controller was generated by Devise 

SCREENSHOTS: 
To start, I made a temporary landing page to set up the routes nicely. 
image

I still don't have the buttons to route them but I will place them soon or in the next PR. 

Admins Log-In 
image

Admins Sign-up --- to be deleted and have it signed up at console
image

User Sign-up
image

User Log-in
image

Forgot Your Password 
image

### CODE REVIEW

File: app/views/landing/signupUser.html.erb [hunk 1]
```diff
@@ -0,0 +1,4 @@
+<h1>Welcome to the Signup page</h1>
+
+# link_to 'Categories List', categories_path|
+# link_to 'Home page', root_path
```
Comment (analyn-cajocson) [inline app/views/landing/signupUser.html.erb]: delete unnecessary comments

Comment (analyn-cajocson) [reply app/views/landing/signupUser.html.erb]: and please rename this file base on rails convention

File: config/routes.rb [hunk 2]
```diff
@@ -1,3 +1,5 @@
 Rails.application.routes.draw do
-  # For details on the DSL available within this file, see https://guides.rubyonrails.org/routing.html
+  devise_for :admins, path: 'admins', controllers: { sessions: "admins/sessions" }
+  devise_for :users, path: 'users', controllers: { sessions: "users/sessions" }
+  root 'landing#signupUser'
```
Comment (analyn-cajocson) [inline config/routes.rb]: please follow rails convention

File: spec/rails_helper.rb [hunk 3]
```diff
@@ -5,37 +5,61 @@
...
-  config.before(:suite) do
-    DatabaseRewinder.clean_all
-  end
+  # You can uncomment this line to turn off ActiveRecord support entirely.
+  # config.use_active_record = false
 
-  config.after(:each) do
-    DatabaseRewinder.clean
-  end
-end
```
Comment (analyn-cajocson) [inline spec/rails_helper.rb]: Better not to delete these codes

File: spec/model/trial_user_spec.rb [hunk 4]
```diff
@@ -0,0 +1,28 @@
+require 'rails_helper'
+
+RSpec.describe 'User', type: :model do
+  it 'is valid with valid attributes' do
+    user1 = User.new(email: 'email@example.com', password: 'password')
+    expect(user1).not_to be_valid
```
Comment (analyn-cajocson) [inline spec/model/trial_user_spec.rb]: This is so confusing, you're test is `is valid with valid attributes` but you're expecting an invalid result

Comment (analyn-cajocson) [reply spec/model/trial_user_spec.rb]: Try to complete the attributes since this test is a happy path test

File: spec/model/trial_user_spec.rb [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+
+  it 'is not valid without email' do
+    user2 = User.new(email: nil)
+    expect(user2).not_to be_valid
+  end
+
+  it 'is not valid without username' do
+    user2 = User.new(username: nil)
+    expect(user2).not_to be_valid
+  end
```
Comment (analyn-cajocson) [inline spec/model/trial_user_spec.rb]: Be specific with your test. If you are testing the validation presence of attribute `last_name` then you should provide the values to the rest of the user's attributes except for `last_name`. Or you may better use this gem https://github.com/thoughtbot/shoulda

File: spec/models/trial_user_spec.rb [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+  it 'is not valid without email' do
+    user2 = User.new(email: nil)
+    expect(user2).not_to be_valid
+  end
+
+  it 'is not valid without username' do
+    user2 = User.new(username: nil)
+    expect(user2).not_to be_valid
+  end
+end
```
Comment (analyn-cajocson) [inline spec/models/trial_user_spec.rb]: Duplicate specs: `spec/model/trial_user_spec.rb`

**Review by analyn-cajocson — CHANGES_REQUESTED** [review]

**Comment by curtydudes** [comment]
This code address the Trader -- User Story 1&2. I also added here the RSPEC Model for users. 

I set up multiple devise accounts following the wiki here for devise: https://github.com/heartcombo/devise/wiki/How-to-Setup-Multiple-Devise-User-Models. Following this, I will create a function where the Admin can only login their credentials as the sign up will happen in the console as per the wiki. 

Models: 
I got to create an admin and user models.

Views: 
Devise automatically created admin and user views while I manually added a landing folder with a temporary landing page called signupUser.html.rb (will be changing this name appropriately)

Controllers: 
Created a landing page temporarily to set up shop and to have some pseudo base
An admins controller and users controller was generated by Devise 

SCREENSHOTS: 
To start, I made a temporary landing page to set up the routes nicely. 
image

I still don't have the buttons to route them but I will place them soon or in the next PR. 

Admins Log-In 
image

Admins Sign-up --- to be deleted and have it signed up at console
image

User Sign-up
image

User Log-in
image

Forgot Your Password 
image

File: config/routes.rb (hunk 2 above)
Comment (analyn-cajocson) [reply config/routes.rb]: @curtydudes please change this

**Review by analyn-cajocson — CHANGES_REQUESTED** [review]

**Review by analyn-cajocson — APPROVED** [review]

**Pull request merged**